commit 1d91e6e7d6f43dc776737b16a19a33caf70da7fa
Author: maralorn <mail@maralorn.de>
Date:   Wed Jun 14 01:12:16 2023 +0200

    maralorns unread hotkey fix

diff --git a/src/components/views/rooms/RoomList.tsx b/src/components/views/rooms/RoomList.tsx
index 06396de03c..7ed9bc7177 100644
--- a/src/components/views/rooms/RoomList.tsx
+++ b/src/components/views/rooms/RoomList.tsx
@@ -517,7 +517,7 @@ export default class RoomList extends React.PureComponent<IProps, IState> {
                 // filter to only notification rooms (and our current active room so we can index properly)
                 listRooms = listRooms.filter((r) => {
                     const state = RoomNotificationStateStore.instance.getRoomState(r);
-                    return state.room.roomId === roomId || state.isUnread;
+                    return state.room.roomId === roomId || state.hasUnreadCount;
                 });
             }
 
